### ISSUE TITLE
BRD wallet import not getting full balance

### ISSUE THREAD

**Issue by jarendy** [BlueWallet/BlueWallet#2594]
Hi BlueWallet Team,

Previously i have this issue:
https://github.com/BlueWallet/BlueWallet/issues/2090
It fixed in 5.6.7 and I successfully import,
and got the same amount as BRD wallet.

Few months later, I just update bluewallet to v6.0.3, it got wrong value again.
It seems the transaction is not updated in BlueWallet.

![3485ab89-4982-4d60-a72e-28b180fc16e0](https://user-images.githubusercontent.com/74204027/107319703-ee151400-6ad1-11eb-9916-b0819863f0e5.jpg)

It said the connection to server is lost.
I try using different wifi / 4g didn't solved the issue.

If I refresh again and wait around 15-30 minutes,
It just show the transaction from 2 months ago.

![WhatsApp Image 2021-02-09 at 12 37 54](https://user-images.githubusercontent.com/74204027/107320780-f0786d80-6ad3-11eb-8c9f-9b6930e3ecbb.jpeg)

Is there any information that you need to investigate?
Appreciate your help!

**Comment by limpbrains** [comment]
Looks like a connection issue to me. I've just re-imported my BRD wallet, and it works fine. 
Are you using custom Electrum server?
Check network connection under Settings -> Network -> Electrum server. Status should show `Connected`
Maybe you should try VPN?

**Comment by jarendy** [comment]
Setting > Network > Electrum Server. Status connected. (electrum1.bluewallet.io:443)

I try using VPN not help.
Then I try to reimport, still the same.

I remember, 2 month ago, I enable segwit in BRD, since segwit will mandatory in BRD in the future.
After that it didn't update. Is it something to do with it?

**Comment by limpbrains** [comment]
> I remember, 2 month ago, I enable segwit in BRD, since segwit will mandatory in BRD in the future.

Yes, probably this is the issue. 
Are you receiving new funds on bech32 addresses ? (addresses started with bc1) 
Can find in BRD first transaction that is mission in BW and check if it is first one with bech32 address?

**Comment by jarendy** [comment]
> Are you receiving new funds on bech32 addresses ? (addresses started with bc1)

Yes.

> Can find in BRD first transaction that is mission in BW and check if it is first one with bech32 address?

The latest transaction using legacy address, 4th December 2020, 03:17 PM.
This is the latest transaction show in BlueWallet.

Then from 8th December 2020, 10:09AM, onward, Im using bc1 address.
This 8th December transaction is the first transaction is missing from BlueWallet.

I try to make sure the value in my legacy wallet too.
In BlueWallet, it show : 0.1532706 BTC
![WhatsApp Image 2021-02-11 at 10 51 01](https://user-images.githubusercontent.com/74204027/107639551-9af2bb00-6ca3-11eb-9bf9-51d8f230f000.jpeg)

But when I try to import my legacy wallet to electrum,
It show : 0.15241943 BTC
![electrum legacy wallet](https://user-images.githubusercontent.com/74204027/107639619-b2ca3f00-6ca3-11eb-8d8f-b977acd6b458.PNG)

It show difference 0.00085117 BTC.

Please correct me if I'm wrong.
What do you think?

**Comment by Overtorment** [comment]
random thought: too small gap limit on bw..?

**Comment by jarendy** [comment]
> random thought: too small gap limit on bw..?

I'm sorry but I didn't get what you mean by that?

**Comment by limpbrains** [comment]
> random thought: too small gap limit on bw..?

I don't think so. Wallet is used on mobile, BRD will not give you new address, unless previous one is used. 

I'm out of ideas for now, can't reproduce on my BRD.

**Comment by Transisto** [comment]
Not sure if this is sorted out but when BRD transitioned to segwit address their change address is completely not standard.

One way to try solving this would be to send the full balance to one of your address on BRD wallet.
* Might have to pay relatively high network fees and lose some UTXO privacy.

**Comment by jarendy** [comment]
> Not sure if this is sorted out but when BRD transitioned to segwit address their change address were completely not standard.
> 
> One way to try solving this would be to send the full balance to one of your address on BRD wallet.
> Might have to pay relatively high network fees and lose some UTXO privacy.

I try electrum wallet. Both legacy and segwit wallet value are correct.
It seems, blue wallet only add legacy wallet and it didn't have correct value. But somehow other can't reproduce the issue.
I'm also not sure what to do next in bluewallet.

**Comment by Overtorment** [comment]
@jarendy are you saying that importing your bradwallet seed into electrum works with no issues? did you have to do it 2 times (in order to see legacy address balance and segwit address balance)?
what are the derivation paths..?

cc @limpbrains

**Comment by jarendy** [comment]
> @jarendy are you saying that importing your bradwallet seed into electrum works with no issues? 
>
yes. after importing to electrum, I test to move btc from segwit wallet to other segwit address succesfully.

> did you have to do it 2 times (in order to see legacy address balance and segwit address balance)?
>
yes, 2 times, 1 for legacy address (p2pkh), 1 one for segwit address (native-segwit / p2wpkh).
both have different value, if I sum both of them, match my brd wallet value.

> what are the derivation paths..?
>
both, m/0'

**Comment by Overtorment** [comment]
is the problem still there?

**Comment by jarendy** [comment]
> is the problem still there?

I believe it still there.

**Comment by limpbrains** [comment]
If anyone can give me a seed backup seed for BRD wallet with this problem, I will try to fix it. Wallet can be empty.

**Comment by jarendy** [comment]
> If anyone can give me a seed backup seed for BRD wallet with this problem, I will try to fix it. Wallet can be empty.

You can create a new BRD legacy wallet account. put small BTC (0.0001) in that legacy address.

Then activate BRD segwit. put another small BTC (0.0001) in that segwit wallet address.

After that import BRD to bluewallet, it will show only 0.0001 not total 0.0002.

**Comment by Overtorment** [comment]
@jarendy this is how @limpbrains initially added support for BRD, so in this rather simple test case everything should work fine I believe

**Comment by jarendy** [comment]
> @jarendy this is how @limpbrains initially added support for BRD, so in this rather simple test case everything should work fine I believe

I just do similar to that simple test case, the different, I got payment from mining each 3-5 days back then.
after 2 month it got out of sync.
I also run out of idea. Maybe it is my account that have problem not bluewallet.

**Comment by Overtorment** [comment]
> n

can you create a wallet from scratch on BRD? reproduce the problem with bluewallet, then move funds (so the wallet has 0 on it) and post mnemonic seed?

**Comment by limpbrains** [comment]
> > If anyone can give me a seed backup seed for BRD wallet with this problem, I will try to fix it. Wallet can be empty.
> 
> You can create a new BRD legacy wallet account. put small BTC (0.0001) in that legacy address.
> 
> Then activate BRD segwit. put another small BTC (0.0001) in that segwit wallet address.
> 
> After that import BRD to bluewallet, it will show only 0.0001 not total 0.0002.

Made this test and received full balance in BlueWallet.

**Comment by jarendy** [comment]
> can you create a wallet from scratch on BRD? reproduce the problem with bluewallet, then move funds (so the wallet has 0 on it) and post mnemonic seed?

> > > If anyone can give me a seed backup seed for BRD wallet with this problem, I will try to fix it. Wallet can be empty.
> > 
> > 
> > You can create a new BRD legacy wallet account. put small BTC (0.0001) in that legacy address.
> > Then activate BRD segwit. put another small BTC (0.0001) in that segwit wallet address.
> > After that import BRD to bluewallet, it will show only 0.0001 not total 0.0002.
> 
> Made this test and received full balance in BlueWallet.

I just update bluewallet then reimport my wallet. Still have the issue. Still same number shown 0.1532706 BTC.

I just remember, when I start and learn mining, I have a friend that have similar conditions to me.
He have BRD wallet, using legacy to safe his mining result, Then he also enable BRD segwit not long after we discuss segwit in BRD. But he never use Bluewallet. I will expain everythig to him and try to ask to him to import to Bluewallet, and please give me some time, I will post the result.

**Comment by jarendy** [comment]
Thanks to my friend, he agree to help to import the wallet temporary to my ipad.
Im using the same ipad mini 2 to import both wallet.
The issue still there.

After successfully import to latest blue wallet on my ipad,
For the first time, it show 61 error code, twice.

![IMG_4325](https://user-images.githubusercontent.com/74204027/128182434-264acfd0-bd7a-4319-8fcd-4877dd9d689d.PNG)

![IMG_4326](https://user-images.githubusercontent.com/74204027/128182491-9590fff5-f81b-4252-9a39-0faefce98003.PNG)

![IMG_4327](https://user-images.githubusercontent.com/74204027/128182504-aa3bf4d5-a0a3-48cc-af3e-814c85c9f424.PNG)

![IMG_4328](https://user-images.githubusercontent.com/74204027/128182514-9f16c780-5a53-4be5-913e-9e677abf341d.PNG)

![IMG_4329](https://user-images.githubusercontent.com/74204027/128182528-ae447b79-991e-49c2-a443-e126b82ac6ce.PNG)

![IMG_4330](https://user-images.githubusercontent.com/74204027/128182543-411c20c8-e55a-47f5-a58b-207b77f0204e.PNG)

![IMG_4331](https://user-images.githubusercontent.com/74204027/128182568-08b1558c-bba1-4a2e-bc09-d51aab88423c.PNG)

![IMG_4332](https://user-images.githubusercontent.com/74204027/128182577-abbb61d2-1166-47f0-a2a7-5a58afdace60.PNG)

![IMG_4333](https://user-images.githubusercontent.com/74204027/128182594-8eabf046-3188-4c6e-8a06-b63420d3d2a9.PNG)

![IMG_4334](https://user-images.githubusercontent.com/74204027/128182599-60d6189a-ab58-43b0-9f45-d54d152ae425.PNG)

![IMG_4335](https://user-images.githubusercontent.com/74204027/128182608-e9f404ed-b24c-4c93-8998-f900939c8d9a.PNG)

![IMG_4336](https://user-images.githubusercontent.com/74204027/128182619-43dba1b8-24e5-43eb-8545-8ca9c4351b97.PNG)

![IMG_4337](https://user-images.githubusercontent.com/74204027/128182630-4e9d3cf6-5de6-4a98-8a11-e852f445e42b.PNG)

![IMG_4338](https://user-images.githubusercontent.com/74204027/128182641-01801466-003d-4226-ac6e-ef8e1f913837.PNG)

![IMG_4339](https://user-images.githubusercontent.com/74204027/128182650-593faa83-d81e-4e08-8015-6bf4cd646dd6.PNG)

After finishing sync, these are the btc value show in blue wallet:
![IMG_4340](https://user-images.githubusercontent.com/74204027/128182729-59652c4f-f04d-4ffa-83f6-2933e2c22c9e.PNG)

Meanwhile when I confirm with Electrum Legacy Wallet.
![electrum legacy wallet 1](https://user-images.githubusercontent.com/74204027/128182920-8144119f-8a71-4e7a-bce8-28365b7b25f8.png)

![electrum legacy wallet 2](https://user-images.githubusercontent.com/74204027/128182932-4718f027-d840-4ecf-a82e-c6bd32c0890f.PNG)

My wallet have difference 0.00085117 BTC
My friend wallet have difference 0.0007633 BTC

If I sum both of wallet,
Electrum Legacy Wallet + Electrum Segwit Wallet = Total in BRD.

Is it possible because of ipad mini 2 have low RAM?

**Comment by jarendy** [comment]
I'm not comfortable to share my xpubs.

Here the detail I remember:
We mine using antminer s9
Using viabtc.com pool
We only mine BTC

From the beginning we using BRD legacy wallet for few months
Then enable BRD Segwit Wallet for few months

After that, we import to Bluewallet on Ipad Mini 2, It show different value.
Meanwhile BRD and Electrum show same value.

I don't mind this issue to be closed.
I really wish you guys the best. Stay safe and stay healthy!
